tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```
@@ -0,0 +1,761 @@
+            if (hashEncodingOption == null) {
+                descriptor.setHashEncoding(Encoding.BASE64);
+            } else {
+                try {
+                    descriptor.setHashEncoding(Encoding.valueOf(hashEncodingOption.toUpperCase()));
+                } catch (IllegalArgumentException | NullPointerException e) {
+                    errorHandler(e);
+                }
+            }
+            if (populateOption == null) {
+                descriptor.setPopulate(true);
+            } else {
+                descriptor.setPopulate(Boolean.valueOf(populateOption));
+            }
+
+            if (levelsOption == null) {
+                descriptor.setLevels(2);
+            } else {
+                try {
+                    descriptor.setLevels(Integer.parseInt(levelsOption));
+                } catch (NumberFormatException e) {
+                    errorHandler(e);
+                }
+            }
+            if (encodedOption == null) {
+                descriptor.setEncoded(true);
+            } else {
+                try {
+                    descriptor.setEncoded(Boolean.parseBoolean(encodedOption));
+                } catch (IllegalArgumentException e) {
+                    errorHandler(e);
+                }
+            }
+            if (createCredentialStore != null) {
+                descriptor.setCreateCredentialStore(Boolean.valueOf(createCredentialStore));
+            } else {
+                descriptor.setCreateCredentialStore(true);
+            }
+
+            if (credentialStoreOption != null) {
+                descriptor.setCredentialStore(credentialStoreOption);
+            }
+
+            if (secretKeyAliasOption != null) {
+                descriptor.setSecretKeyAlias(secretKeyAliasOption);
+            } else {
+                descriptor.setSecretKeyAlias("key");
+            }
+            descriptors.add(descriptor);
```

`findMissingRequiredValuesAndSetValues` is not applied in this case, hence it's not clear what is missing. For example the following ends with no identities converted (silently skipped by the initial check in `createFileSystemRealm`?)

```
$ ./bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configuration/fs-realm-users-plain -o ./standalone/configuration/fs-realm-users-plain-enc --debug
Creating encrypted realm for: ./standalone/configuration/fs-realm-users-plain
```

---

Added a warning message to indicate something is missing. Although now it should flag it before entering that method

https://github.com/wildfly-security/wildfly-elytron/pull/1676/files#diff-ebe53241afd70e1e5743d6db5b39150461c335abce6c083cafbf6aae4b4c1e2dR664-R666